app/express/interceptors/session/SessionStore.js

```diff
@@ -0,0 +1,129 @@
+
+const {
+  Logger,
+} = require('@dojot/microservice-sdk');
+
+const logger = new Logger('backstage:express/interceptors/session/SessionStore');
+
+
+module.exports = (session) => {
+  const { Store } = session;
+
+  /**
+   * Class to be used with session-express, see more about Sessions Stores
+   * at https://github.com/expressjs/session#session-store-implementation
+   *
+   * NOTE: This class implementation is based on
+   * https://github.com/tj/connect-redis/tree/v5.1.0 which has an MIT license.
+   *
+   */
+  class SessionStore extends Store {
+    constructor(options = {}, redis, keycloak) {
+      super(options);
+      this.redisManagement = redis.getManagementInstance();
+      const logoutBound = keycloak.getRequestsInstance()
+        .logout.bind(keycloak.getRequestsInstance());
+      this.redisManagement.setFuncToCallBeforeDestroy(logoutBound);
+    }
+
+    /**
+     * This  method is used to get a session from the store
+     * given a session ID (sid). The callback should be called as
+     * callback(error, session).
+     *
+     * The session argument should be a session if found,
+     * otherwise null or undefined if the session
+     * was not found (and there was no error).
+     * A special case is made when error.code === 'ENOENT'
+     * to act like callback(null, null).
+     *
+     * @param {string} sid  session ID
+     * @param {function} cb
+     * @returns
+     */
+    async get(sid, cb = () => {}) {
+      logger.debug(`get: sid=${sid}`);
+      try {
+        const data = await this.redisManagement.get(sid);
+        return cb(null, data);
+      } catch (err) {
+        logger.warn(`get: err=${err.stack || err}`);
+        return cb(err);
+      }
+    }
+
+    /**
+     * This method is used to set a session into the store given a
+     * session ID  and session  object.
+     *
+     * The callback should be called as callback(error)
+     * once the session has been set in the store.
+     *
+     * @param {string} sid session ID
+     * @param {Object} sess session
+     * @param {function} cb
+     * @returns
+     */
+    async set(sid, sess, cb = () => {}) {
+      logger.debug(`set: sid=${sid}`);
+      try {
+        await this.redisManagement.set(sid, sess);
+        return cb(null, 'OK');
+      } catch (err) {
+        logger.warn(`set: err=${err.stack || err}`);
+        return cb(err);
+      }
+    }
```

Comment:
Did the `async/await` statement work correctly?
I ask this because it seems to me that the `express-session` module does not support it, so in this case I was in doubt if we should use:
```javascript
this.redisManagement.set(sid, sess)
    .then( () => cb(null, 'OK') )
    .catch(...)
```
Nodejs must know how to solve this correctly, right? 
It's more of a curiosity actually :smile:

Reply:
It works correctly, but honestly I would need to study and understand better why.